Mobile Stories block (part2: on done) (#25771)
WordPress/gutenberg: packages/block-editor/src/components/story-update-progress/README.md

StoryUpdateProgress
===================

> `StoryUpdateProgress` shows a progress bar while the media files associated with a Story block are being saved and uploaded

Reviewer: Same than what I mentioned [here.](https://github.com/WordPress/gutenberg/pull/25242/files#discussion_r501515233)

It's possible that this component could give us troubles, since it's defined on the `gutenberg` repo but used only on a `jetpack` block and referring to a jetpack/wpcom feature (Stories).

What I would recommend in this case is to make all required bridge methods generic, and move this component to Jetpack.

Pinging @hypest and @maxme to get their opinion, since this would be a bigger refactor.

Author: As discussed elsewhere with @etoledom - I'll make this component generic given it may be used on other blocks containing a collection of media in the future, and that will solve this situation 👍

Author: Addressed in the set of PRs mentioned / starting with this https://github.com/WordPress/gutenberg/pull/26005

## Usage

Usage example

```jsx
import { ImageBackground, Text, View } from 'react-native';
import {
	StoryUpdateProgress,
} from '@wordpress/block-editor';

function StoryProgress( { url, id } ) {
	return (
		<StoryUpdateProgress
			mediaId={ id }
			renderContent={ ( { isSaveFailed, retryMessage } ) => {
				return (
					<ImageBackground
						resizeMethod="scale"
						source={ { uri: url } }
					>
						{ isSaveFailed &&
							<View>
								<Text>{ retryMessage }</Text>
							</View>
						}
					</ImageBackground>
				);
			} }
		/>
	);
}
```

## Props

### mediaFiles

> A collection of media ID that identifies the current story upload.

Reviewer 2: ```suggestion
A collection of media IDs that identifies the current story upload.
```

> A collection of media ID that identifies the current story upload.

Reviewer 2: ```suggestion
A collection of media IDs that identify the current story upload.
```

- Type: `Array`
- Required: Yes
- Platform: Mobile

### renderContent

Content to be rendered along with the progress bar, usually the thumbnail of the media being uploaded.

- Type: `React components`
- Required: Yes
- Platform: Mobile

It passes an object containing the following properties:

`{ isUploadInProgress, isUploadFailed, isSaveInProgress, isSaveFailed, retryMessage }`

### onUpdateMediaProgress

Callback called when the progress of the upload is updated.

- Type: `Function`
- Required: No
- Platform: Mobile

The argument of the callback is an object containing the following properties:

`{ mediaId, mediaUrl, progress, state }`

### onFinishMediaUploadWithSuccess

Callback called when the media file has been uploaded successfully.

- Type: `Function`
- Required: No
- Platform: Mobile

The argument of the callback is an object containing the following properties:

`{ mediaId, mediaServerId, mediaUrl, progress, state }`

### onFinishMediaUploadWithFailure

Callback called when the media file couldn't be uploaded.

- Type: `Function`
- Required: No
- Platform: Mobile

The argument of the callback is an object containing the following properties:

> `{ mediaId, progress, state }`

Reviewer 2: @mzorz since the upload failed do we still want to pass the progress and the state? Just checking 🙏

Author: this part is copied from the MediaUploadProgress component (we may replace that one with this one in the long run as mentioned here https://github.com/WordPress/gutenberg/pull/26005#issuecomment-707667916)

I think we need it so it's easy to override the progress bar being updated / reset if needed (up to the caller to send 0.0 or 1.0, for example)

### onMediaUploadStateReset

> Callback called when the media upload is reset

Reviewer 2: ```suggestion
Callback called when the media upload is reset.
```
